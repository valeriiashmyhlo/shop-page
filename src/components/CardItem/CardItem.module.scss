.cardSoldOverlay {
  opacity: 0;
  position: absolute;
  top: 0;
  bottom: 0;
  color: white;
  font-size: 18px;
  font-weight: 600;
  background: rgba(black, .45);
  text-transform: uppercase;
  text-align: center;
  padding-top: 60px;

  &.isSold {
    opacity: 1;
  }
}

.cardPrice {
  font-weight: 600;
  margin-bottom: 0;
}

.cardLikeBtn {
  position: absolute;
  top: 10px;
  right: 10px;
  color: grey;
  border-color: grey;

  &:hover, &:focus {
    color: grey;
    border-color: grey;
  }

  &.liked {
    color: red;
    border-color: red;
  }
}
