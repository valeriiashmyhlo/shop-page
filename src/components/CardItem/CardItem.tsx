import React, { FC } from "react";
import { Card, Button } from "antd";
import { LikeOutlined } from "@ant-design/icons";
import classNames from "classnames";
import styles from "./CardItem.module.scss";

const FallbackImg =
  "https://www.enmp.edu.do/wp-content/plugins/pixelthrone__vc_extension/assets/img/shortcode-frontend/empty-image-thumb.png";

export const CardItem: FC<{
  brand: string;
  id: string;
  img: string;
  name: string;
  price: string;
  sold: boolean;
  isLiked: boolean;
  onLike: (id: string) => void;
}> = ({ id, name, img, brand, price, sold, isLiked, onLike }) => {
  return (
    <Card
      size="small"
      style={{ width: 220, margin: "10px" }}
      cover={
        <>
          <img
            alt="productImg"
            src={FallbackImg}
            // src={img}
            // If believe, there should be src={img} but it doesn't seems right, since these are the images of people, not the products
          />
          <div
            className={classNames(styles.cardSoldOverlay, {
              [styles.sold]: sold,
            })}
          >
            sold
          </div>
        </>
      }
    >
      <Button
        onClick={() => onLike(id)}
        className={classNames(styles.cardLikeBtn, {
          [styles.liked]: isLiked,
        })}
        size="small"
        icon={<LikeOutlined />}
        data-testid="like-btn"
      />
      <p>{name}</p>
      <p>{brand}</p>
      <p className={styles.cardPrice}>{price}</p>
    </Card>
  );
};
