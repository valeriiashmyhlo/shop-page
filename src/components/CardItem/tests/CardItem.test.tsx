import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { CardItem } from "../CardItem";

describe(`<CardItem /> tests`, () => {
  const props = {
    brand: "brandName",
    id: "1",
    img: "imgUrl",
    name: "name",
    price: "33.33",
    sold: false,
    isLiked: false
  };

  it("should render CardItem with correct props", () => {
    const { container } = render(<CardItem onLike={() => {}} {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should mark sold items as sold", () => {
    const { container } = render(<CardItem onLike={() => {}} {...props } sold isLiked />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call onLike", () => {
    const onLike = jest.fn();
    const { getByTestId } = render(<CardItem onLike={onLike} {...props} />);
    fireEvent.click(getByTestId("like-btn"));
    expect(onLike).toHaveBeenCalledWith("1");
    expect(onLike).toHaveBeenCalledTimes(1);
  });
});
