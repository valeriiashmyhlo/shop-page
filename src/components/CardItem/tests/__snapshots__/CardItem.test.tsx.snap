// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem /> tests should mark sold items as sold 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-small"
  style="width: 220px; margin: 10px;"
>
  <div
    class="ant-card-cover"
  >
    <img
      alt="productImg"
      src="https://www.enmp.edu.do/wp-content/plugins/pixelthrone__vc_extension/assets/img/shortcode-frontend/empty-image-thumb.png"
    />
    <div
      class="cardSoldOverlay sold"
    >
      sold
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    <button
      class="ant-btn ant-btn-sm ant-btn-icon-only cardLikeBtn liked"
      data-testid="like-btn"
      type="button"
    >
      <span
        aria-label="like"
        class="anticon anticon-like"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="like"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M885.9 533.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.4-65.5-111.1a67.67 67.67 0 00-34.3-9.3H572.4l6-122.9c1.4-29.7-9.1-57.9-29.5-79.4A106.62 106.62 0 00471 99.9c-52 0-98 35-111.8 85.1l-85.9 311H144c-17.7 0-32 14.3-32 32v364c0 17.7 14.3 32 32 32h601.3c9.2 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7-.2-12.6-2-25.1-5.6-37.1zM184 852V568h81v284h-81zm636.4-353l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H329V564.8l99.5-360.5a44.1 44.1 0 0142.2-32.3c7.6 0 15.1 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.2 32.1-19.6 43z"
          />
        </svg>
      </span>
    </button>
    <p>
      name
    </p>
    <p>
      brandName
    </p>
    <p
      class="cardPrice"
    >
      33.33
    </p>
  </div>
</div>
`;

exports[`<CardItem /> tests should render CardItem with correct props 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-small"
  style="width: 220px; margin: 10px;"
>
  <div
    class="ant-card-cover"
  >
    <img
      alt="productImg"
      src="https://www.enmp.edu.do/wp-content/plugins/pixelthrone__vc_extension/assets/img/shortcode-frontend/empty-image-thumb.png"
    />
    <div
      class="cardSoldOverlay"
    >
      sold
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    <button
      class="ant-btn ant-btn-sm ant-btn-icon-only cardLikeBtn"
      data-testid="like-btn"
      type="button"
    >
      <span
        aria-label="like"
        class="anticon anticon-like"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="like"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M885.9 533.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.4-65.5-111.1a67.67 67.67 0 00-34.3-9.3H572.4l6-122.9c1.4-29.7-9.1-57.9-29.5-79.4A106.62 106.62 0 00471 99.9c-52 0-98 35-111.8 85.1l-85.9 311H144c-17.7 0-32 14.3-32 32v364c0 17.7 14.3 32 32 32h601.3c9.2 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7-.2-12.6-2-25.1-5.6-37.1zM184 852V568h81v284h-81zm636.4-353l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H329V564.8l99.5-360.5a44.1 44.1 0 0142.2-32.3c7.6 0 15.1 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.2 32.1-19.6 43z"
          />
        </svg>
      </span>
    </button>
    <p>
      name
    </p>
    <p>
      brandName
    </p>
    <p
      class="cardPrice"
    >
      33.33
    </p>
  </div>
</div>
`;
